File: server/router/data_report/data_report.go
package data_report

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type DataReportRouter struct {
}

func (s *DataReportRouter) InitApiRouter(Router *gin.RouterGroup) {
	apiRouter := Router.Group("data-report")
	apiRouter.Use(middleware.OperationRecord())
	{
		dayOverviewGroup := apiRouter.Group("day-overview") // 每日总览
		{
			dayOverviewGroup.POST("list", dayOverviewApi.List)
		}

		retentionStatusGroup := apiRouter.Group("retention-status") // 留存情况
		{
			retentionStatusGroup.POST("list", retentionStatusApi.List)
		}
	}
}

File: web/src/api/dataReport.js
import service from '@/utils/request'

// @Router /data-report/day-overview/list [post]
export const dayOverviewList = (data) => {
    return service({
        url: '/data-report/day-overview/list',
        method: 'post',
        data
    })
}

// @Router /data-report/retention-status/list [post]
export const retentionStatusList = (data) => {
    return service({
        url: '/data-report/retention-status/list',
        method: 'post',
        data
    })
}

// @Router /data-report/payment-status/list [post]
export const paymentStatusList = (data) => {
    return service({
        url: '/data-report/payment-status/list',
        method: 'post',
        data
    })
}